src/osmet/mod.rs

```diff
@@ -93,10 +93,17 @@ pub fn pack_osmet(config: PackOsmetConfig) -> Result<()> {
     // First, mount the two main partitions we want to suck out data from: / and /boot. Note
     // MS_RDONLY; this also ensures that the partition isn't already mounted rw elsewhere.
     let disk = Disk::new(&config.device)?;
-    let boot = disk.mount_partition_by_label("boot", mount::MsFlags::MS_RDONLY)?;
-    let root = disk.mount_partition_by_label("root", mount::MsFlags::MS_RDONLY)?;
-
     // now, we do a first scan of the boot partition and pick up files over a certain size
+    let boot;
+    let root;
+    if !config.root.trim().is_empty() && !config.boot.trim().is_empty() {
+        boot = disk.mount_partition_by_label(&config.boot, false, mount::MsFlags::empty())?;
+        root = disk.mount_partition_by_label(&config.root, false, mount::MsFlags::empty())?;
```

Comment:
Hmm, I think it's confusing to have `mount_partition_by_label` take either a label _or_ a full device path.